test/unit/scratchOrgInfoApiTest.ts

```diff
@@ -0,0 +1,539 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as sinon from 'sinon';
+import { assert, expect } from 'chai';
+import { env, Duration } from '@salesforce/kit';
+import { stubMethod } from '@salesforce/ts-sinon';
+import { SObject } from 'jsforce';
+import { shouldThrow } from '../../src/testSetup';
+import { Org } from '../../src/org';
+import { Connection } from '../../src/connection';
+import { AuthInfo } from '../../src/authInfo';
+import { MyDomainResolver } from '../../src/status/myDomainResolver';
+import SettingsGenerator from '../../src/scratchOrgSettingsGenerator';
+import {
+  requestScratchOrgCreation,
+  ScratchOrgInfo,
+  JsForceError,
+  deploySettingsAndResolveUrl,
+  pollForScratchOrgInfo,
+  authorizeScratchOrg,
+} from '../../src/scratchOrgInfoApi';
+import { Messages } from '../../src/messages';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/core', 'scratchOrgInfoApi');
+const errorCodesMessages = Messages.loadMessages('@salesforce/core', 'scratchOrgErrorCodes');
+
+const TEMPLATE_SCRATCH_ORG_INFO: ScratchOrgInfo = {
+  LoginUrl: 'https://login.salesforce.com',
+  Snapshot: '1234',
+  AuthCode: '1234',
+  Status: 'New',
+  SignupEmail: 'sfdx-cli@salesforce.com',
+  SignupUsername: 'sfdx-cli',
+  Username: 'sfdx-cli',
+  SignupInstance: 'http://salesforce.com',
+};
+
+describe('requestScratchOrgCreation', () => {
+  const sandbox = sinon.createSandbox();
+  const connectionStub = sinon.createStubInstance(Connection);
+
+  beforeEach(() => {
+    stubMethod(sandbox, Org, 'create').resolves(Org.prototype);
+    connectionStub.sobject.withArgs('ScratchOrgInfo').returns({
+      create: sinon.stub().resolves(true),
+    } as unknown as SObject<unknown>);
+    stubMethod(sandbox, Org.prototype, 'getConnection').returns(connectionStub);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('requestScratchOrgCreation', async () => {
```

Comment:
NIT: I'm confused on what this is exactly testing, could you expand the test title

Reply:
It's a test for a basic test on a fully populated `ScratchOrgInfo` object, I added extra comments